front/lib/api/assistant/conversation.ts

```diff
@@ -175,6 +183,119 @@ export async function* postUserMessage(
+  const agentMessageEvents: AgentMessageNewEvent[] = [];
+
+  await front_sequelize.transaction(async (t) => {
+    let nextMessageRank =
+      ((await AssistantMessage.max<number | null, AssistantMessage>("rank", {
+        where: {
+          assistantConversationId: conversation.id,
+        },
+        transaction: t,
+      })) ?? -1) + 1;
+    const userMessage = await AssistantMessage.create(
+      {
+        sId: generateModelSId(),
+        rank: nextMessageRank++,
+        assistantConversationId: conversation.id,
+        parentId: null,
+        assistantUserMessageId: (
+          await AssistantUserMessage.create(
+            {
+              message: message,
+              userContextUsername: context.username,
+              userContextTimezone: context.timezone,
+              userContextFullName: context.fullName,
+              userContextEmail: context.email,
+              userContextProfilePictureUrl: context.profilePictureUrl,
+              userId: user ? user.id : null,
+            },
+            { transaction: t }
+          )
+        ).id,
+      },
+      {
+        transaction: t,
+      }
+    );
+    userMessageEvent = {
+      type: "user_message_new",
+      message: {
+        id: userMessage.id,
+        sId: userMessage.sId,
+        type: "user_message",
+        visibility: "visible",
+        version: 0,
+        parentMessageId: null,
+        user: user,
+        mentions: mentions,
+        message: message,
+        context: context,
+      },
+    };
+
+    // for each assistant mention, create an "empty" agent message
+    for (const m of mentions) {
+      if (isAssistantAgentMention(m)) {
+        const agentMessage = await AssistantMessage.create(
+          {
+            sId: generateModelSId(),
+            rank: nextMessageRank++,
+            assistantConversationId: conversation.id,
+            parentId: userMessage.id,
+            assistantAgentMessageId: (
+              await AssistantAgentMessage.create({}, { transaction: t })
+            ).id,
+          },
+          {
+            transaction: t,
+          }
+        );
+
+        agentMessageEvents.push({
+          type: "agent_message_new",
+          created: Date.now(),
+          configurationId: m.configurationId,
+          message: {
+            id: agentMessage.id,
+            sId: agentMessage.sId,
+            type: "agent_message",
+            visibility: "visible",
+            version: 0,
+            parentMessageId: userMessage.sId,
+            status: "created",
+            action: null,
+            message: null,
+            feedbacks: [],
+            error: null,
+            configuration: {
+              sId: m.configurationId,
+              status: "active",
+              name: "foo", // TODO
+              pictureUrl: null, // TODO
+              action: null, // TODO
+              message: null, // TODO
+            },
+          },
+        });
+      }
+    }
+  });
```

Comment:
If the commit  fails, we consider this an unrecoverable error if we're not handling it. I guess that is fine, we'll see in practice I don't expect there'll be too much race here as the execution will be fast